core/store/src/trie/shard_tries.rs

```diff
@@ -35,23 +37,26 @@ impl ShardTries {
         let shards: Vec<_> = (0..num_shards)
             .map(|shard_id| ShardUId { version: shard_version, shard_id: shard_id as u32 })
             .collect();
-        ShardTries {
+        ShardTries(Arc::new(ShardTriesInner {
             store,
-            caches: Arc::new(Self::get_new_cache(&shards)),
-            view_caches: Arc::new(Self::get_new_cache(&shards)),
-        }
+            caches: RwLock::new(Self::get_new_cache(&shards)),
+            view_caches: RwLock::new(Self::get_new_cache(&shards)),
+        }))
+    }
+
+    pub fn is_same(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.0, &other.0)
```

Comment:
Probably better to implement `PartialEq` for `ShardTries`?

Reply:
I'd rather not: https://github.com/near/nearcore/pull/4817#discussion_r708279480

`PartialEq` signals -- "hey, you are supposed to `==` this relatively frequently". For our use-case, we have only a single debug-assertion that verifies identity -- it's better if that use-case stands out.

In a similar vein, if you have only a *single* place where you convert `Foo` to `Bar` it might be better to add an inherent method for that, instead of implementing a blanket `From<Foo> for Bar`. Single use-case is probably contextual (it makes sense to Foo -> Bar in *this specific* context), but the impl is global (under any circumstances, it's ok to turn Foo into Bar).

Reply:
I see. Thanks for the explanation!